src/main/java/com/exactpro/th2/check1/Check1Handler.java

```diff
@@ -68,16 +77,18 @@ public void createCheckpoint(CheckpointRequest request, StreamObserver<Checkpoin
 
     @Override
     public void submitCheckRule(CheckRuleRequest request, StreamObserver<CheckRuleResponse> responseObserver) {
+        var ruleId = request.getStoreResult() ? resultsStorage.createId() : 0;
         try {
             if (logger.isInfoEnabled()) {
                 logger.info("Submit CheckRule request: " + MessageUtils.toJson(request));
             }
 
             CheckRuleResponse.Builder response = CheckRuleResponse.newBuilder();
             try {
-                ChainID chainID = collectorService.verifyCheckRule(request);
+                ChainID chainID = collectorService.verifyCheckRule(request, ruleId);
                 response.setChainId(chainID)
                         .setStatus(RequestStatus.newBuilder().setStatus(SUCCESS));
+                if (ruleId != 0) response.setRuleId(ruleId);
```

Comment:
It won't make any difference because the ID is an int in grpc message. So, it is 0 by default